src/main/java/com/minecolonies/coremod/event/EventHandler.java

```diff
@@ -754,5 +755,8 @@ public static void onWorldUnload(@NotNull final WorldEvent.Unload event)
             ItemBlockHut.checkResearch(null);
             Log.getLogger().info("Removed all colony views");
         }
+        // Since there doesn't seem to be a Forge event for a datapack being unloaded, there's not many better places to handle this.
+        // Otherwise, loading a world with any custom research will show research effects that don't exist on other worlds in the same instance.
+        IResearchEffectRegistry.getInstance().resetRegistry();
```

Comment:
We're doing something similar for recipes in the RecipeManager. We should use a common approach for this problem.